code/modules/mob/living/simple_animal/abnormality/he/siren.dm

```diff
@@ -0,0 +1,153 @@
+	var/list/unsafe = list()
+//SFX
+	var/datum/looping_sound/siren_musictime/musictime
+	var/playstatus = 0
+	var/playrange = 40
+
+//Insight work causes the user to have every buff and debuff removed
+/mob/living/simple_animal/hostile/abnormality/siren/AttemptWork(mob/living/carbon/human/user, work_type)
+	if(work_type == ABNORMALITY_WORK_INSIGHT)
+		work_effect = TRUE
+		if(!playstatus)
+			playsound(loc, 'sound/abnormalities/siren/backtherebenjamin.ogg', 50, FALSE)
+			playstatus = 1 //prevents song overlap
+			addtimer(CALLBACK(src, .proc/stopPlaying), 59 SECONDS)
+			icon_state = "siren_breach"
+	else
+		work_effect = FALSE
+	return TRUE
+
+/mob/living/simple_animal/hostile/abnormality/siren/Life()
+	. = ..()
+	if(meltdown_cooldown < world.time && !datum_reference.working && !playstatus) // Doesn't decrease while working but will afterwards
+		meltdown_cooldown = world.time + meltdown_cooldown_time
+		datum_reference.qliphoth_change(-1)
+		meltdown_imminent = FALSE
+
+	if(datum_reference.qliphoth_meter == 1 && !meltdown_imminent)
+		meltdown_imminent = TRUE
+		playsound(src, 'sound/abnormalities/siren/burningmemory.ogg', 100, FALSE, 40, falloff_distance = 20)
+		playstatus = 1
+		addtimer(CALLBACK(src, .proc/stopPlaying), 52 SECONDS)
+		icon_state = "siren_breach"
+		warning()
+
+	if(song_cooldown < world.time && !datum_reference.qliphoth_meter)
+		if(!musictime)
+			musictime = new(list(src), TRUE)
+		SSlobotomy_corp.InitiateMeltdown(round(SSlobotomy_corp.qliphoth_meltdown_amount/3)+1, TRUE)
+		song_cooldown = world.time + song_cooldown_time
+		playstatus = 1
+	if(playstatus && datum_reference.qliphoth_meter <= 1)
+		warning()
+
+//All it takes is someone to turn it off, either manually
+/mob/living/simple_animal/hostile/abnormality/siren/funpet()
+	if(playstatus && !datum_reference.qliphoth_meter)
+		datum_reference.qliphoth_change(3)
+		stopPlaying()
+		return
+
+//Or by working
+/mob/living/simple_animal/hostile/abnormality/siren/PostWorkEffect(mob/living/carbon/human/user, work_type, pe)
+	if(work_effect)
+		turnBackTime(user)
+	datum_reference.qliphoth_change(5)
+	stopPlaying()
+	return
+
+/mob/living/simple_animal/hostile/abnormality/siren/proc/turnBackTime(mob/living/carbon/human/user)
+	var/mob/living/carbon/human/H = user
+	var/statusChecker = 0
+	var/currentage = H.age
+	var/message
+	if(H in unsafe && prob(25)) //first work is a freebie, subsequent works are risky.
+		to_chat(user, "<span class='danger'>The last thing you remember is your heart stopping.</span>")
+		playsound(loc, 'sound/magic/clockwork/ratvar_attack.ogg', 50, TRUE)
+		H.dust()
+		return
+	H.age = (rand(68) + 17) //minium age is 17, max is 85
+	if (H.age > currentage)
+		message += "You feel older"
+	else if (H.age < currentage)
+		message += "You feel younger"
+	else
+		to_chat(H, "<span class='warning'>Doesn't seem like it did anything this time.</span>")
+		return
+
+	for(var/datum/status_effect/V in H.status_effects)
+		var/datum/status_effect/E = V
```

Comment:
Any reason for this?

Reply:
I'm getting rid of this code, so it shouldn't be an issue anymore. It was far too buggy.